junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java

```diff
@@ -166,11 +242,34 @@ private String determineFullyQualifiedClassName(Path baseDir, String basePackage
 		// @formatter:on
 	}
 
+	/**
+	 * The fully qualified resource name is a {@code /}-separated path.
+	 * <p>
+	 * The path is relative to the classpath root in which the resource is located.
+
+	 * @return the resource name; never {@code null}
+	 */
+	private String determineFullyQualifiedResourceName(Path baseDir, String basePackageName, Path resourceFile) {
+		// @formatter:off
+		return Stream.of(
```

Comment:
While this very readable, 29% of the time in this is spend on joining strings.

![image](https://github.com/junit-team/junit5/assets/6946919/9ae469a4-f788-46b1-8deb-8d946ec69f42)

Reply:
Doing this with a string builder only removes ~30ms, not quite worth it.